tests/test_worker.py

```diff
@@ -607,6 +607,20 @@ def test_deleted_jobs_arent_executed(self):
         # Should not have created evidence of execution
         self.assertEqual(os.path.exists(SENTINEL_FILE), False)
 
+    @slow
+    def test_max_idle_time(self):
+        q = Queue()
+        w = Worker([q])
+        q.enqueue(say_hello, args=('Frank',))
+        result = w.dequeue_job_and_maintain_ttl(1)
+        assert result is not None
+
+        result = w.dequeue_job_and_maintain_ttl(None)
+        assert result is None
+
+        result = w.dequeue_job_and_maintain_ttl(1, max_idle_time=1)
```

Comment:
Can we structure this test such that the `max_idle_time` logic is tested?

If we make the dequeue TTL 1 second and `max_idle_time` 3 seconds, then the worker will attempt to dequeue 3 times before exiting, right?

Reply:
right, clarified the test a bit, let me know if it makes sense